src/Relationships/EntitiesCollection.php

```diff
@@ -95,23 +90,61 @@ protected function &variables(array $requiredItems = null) {
             foreach ($requiredItems as $key => $requiredItem) if (isset($variables[$requiredItem])) unset($requiredItems[$key]);
         }
 
-        if (!$this->entitiesAreLoaded AND ((\count($requiredItems) !== 0) OR $requiredItems === null)) {
-            $this->entitiesAreLoaded = $requiredItems === null;
+        if (!$this->entitiesAreLoaded AND ($requiredItems === null OR (\is_array($requiredItems) AND (boolean) $requiredItems))) {
             $this->loadEntities($requiredItems);
         }
 
         return parent::variables();
     }
 
     /**
+     * @return array
+     */
+    public function asArray(array $requiredItems = null) {
+        return $this->variables($requiredItems);
+    }
+
+    /**
+     * @param mixed $value
+     * @param bool $requiredToLoadCollection
+     * @return bool
+     */
+    public function containsValue($value, $requiredToLoadCollection = true) {
+        return \array_search($value, $this->variables($requiredToLoadCollection ? null : [])) === false ? false : true;
```

Comment:
and isset even faster: https://stackoverflow.com/questions/8401003/which-is-faster-array-key-exists-or-array-search